packages/compass-connect/src/modules/telemetry.js

```diff
@@ -0,0 +1,95 @@
+import { constantCase } from 'constant-case';
+import { createLoggerAndTelemetry } from '@mongodb-js/compass-logging';
+import { isAtlas, isLocalhost, isDigitalOcean } from 'mongodb-build-info';
+import { getCloudInfo } from 'mongodb-cloud-info';
+import ConnectionString from 'mongodb-connection-string-url';
+
+const { track, debug } = createLoggerAndTelemetry('COMPASS-CONNECT-UI');
+
+async function getConnectionData(connectionInfo) {
+  const {connectionOptions: {connectionString, sshTunnel}} = connectionInfo;
+  const connectionStringData = new ConnectionString(connectionString);
+  const hostName = connectionStringData.hosts[0];
+  const { isAws, isAzure, isGcp } = await getCloudInfo(hostName).catch((err) => {
+    debug('getCloudInfo failed', err);
+    return {};
+  });
+  const isPublicCloud = isAws || isAzure || isGcp;
+  const publicCloudName = isAws ? 'AWS' : isAzure ? 'Azure' : isGcp ? 'GCP' : '';
+  const authMechanism = connectionStringData.searchParams.get('authMechanism');
+  const authType = authMechanism
+    ? authMechanism
+    : connectionStringData.username
+      ? 'SCRAM-SHA-1'
+      : 'DEFAULT';
+
+  return {
+    is_localhost: isLocalhost(hostName),
+    is_atlas: isAtlas(hostName),
+    is_public_cloud: !!isPublicCloud,
+    is_do: isDigitalOcean(hostName),
+    public_cloud_name: publicCloudName,
+    auth_type: constantCase(authType),
+    is_ssh_tunnel: !!sshTunnel,
```

Comment:
This is new, right, and not part of the previous telemetry we had? In that case it might be good to already account for what we have in the upcoming [Unified connection layer](https://docs.google.com/document/d/11Q_L-5z3cFen-bUP8natn0rSWcsfniq8oAP1cFxYDy4/edit) work:

> We add tracking to our telemetry infrastructure to track whether the user is connecting using an SSH tunnel/using manual Socks5/not using any kind of proxy, and whether the user is connected to a single-endpoint setup or a replica set.

So maybe for now this could be `tunnel: 'ssh' | 'none'` or `'proxy: 'ssh' | 'none'`?